fn_falcon_sandbox/fn_falcon_sandbox/components/falcon_sandbox_submit_url.py

```diff
@@ -0,0 +1,180 @@
+                    )
+
+            client = self.rest_client()
+            time_now = str(time.time())
+            request_common = RequestsCommon(self.options, kwargs)
+
+            ## preparing inputs
+            falcon_sandbox_environment_id = get_environment_id(
+                falcon_sandbox_environment
+            )
+            falcon_sandbox_action_script = get_runtime_action_script(
+                falcon_sandbox_action_script
+            )
+
+            ## Prepare for submission
+            url, http_method = HA_REST_API_URLS.get("submit_url")
+            url = url.format(API_HOST)
+            form_data = {"url": falcon_sandbox_url}
+
+            for p in HA_LIST_OF_RUNTIME_PARAMS_SUBMIT_URL:
+                k = p.split("falcon_sandbox_")[1]
+                v = eval(p)
+                form_data[k] = v
+
+            submit_header = falcon_sandbox_request_header(API_KEY)
+            submit_header["content-type"] = HA_SUBMIT_URL_CONTENT_TYPE
+            ## Submit url to falcon sandbox
+            yield StatusMessage("Submitting...")
+            response = request_common.execute_call(
+                http_method, url, form_data, log=log, headers=submit_header
+            )
+            job_id = response["job_id"]
+            url, http_method = HA_REST_API_URLS.get("get_state")
+            url = url.format(API_HOST, job_id)
+            yield StatusMessage("Successfully Submitted. \n Job ID: {}".format(job_id))
+
+            ## Get Status of Analysis
+            req_header = falcon_sandbox_request_header(API_KEY)
+            time_running = 0
+            while True:
+                scan_status = request_common.execute_call(
+                    http_method, url, log=log, headers=req_header
+                )
+                status = scan_status["state"]
+                if status in HA_COMPLETED_STATUSES:
+                    yield StatusMessage("Scan Completed with status {}".format(status))
+                    break
+                # Timeout condition
+                if FETCH_REP_TIMEOUT <= (time_running * 60):
+                    yield FunctionError(
+                        "Function Timeout after {} mins".format(time_running)
+                    )
+                    break
+
+                time.sleep(FETCH_REP_STAT_INTRVL)
+                time_running = time_running + (FETCH_REP_STAT_INTRVL/60)
+                yield StatusMessage(
+                    "Scan running as {}. Time Elapsed {} mins".format(
+                        status, time_running
+                    )
+                )
+
+            ## Get scan report summary
+            url, http_method = HA_REST_API_URLS.get("get_summary")
+            url = url.format(API_HOST, job_id)
+            scan_report = request_common.execute_call(
+                http_method, url, log=log, headers=req_header
+            )
+
+            ## Return result to resilient
+            pb = ResultPayload("falcon_sandbox_submit_url", **kwargs)
```

Comment:
Need a comment from @sjcurtinibm here too, to confirm my logic is right.
I'm thinking this ResultPayload should be nearer to the top of the file. As I think one of the `metrics` in the payload is the execution time, which if this is at the end of the file will be tiny.

Reply:
I agree with you.

Reply:
Agree too